target/linux/ramips/dts/mt7628an_tplink_tl-wr902ac-v4.dts

```diff
@@ -0,0 +1,112 @@
+#include "mt7628an_tplink_8m.dtsi"
+#include <dt-bindings/leds/common.h>
+
+/ {
+	compatible = "tplink,tl-wr902ac-v4", "mediatek,mt7628an-soc";
+	model = "TP-Link TL-WR902AC v4";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		button-reset {
+			label = "reset";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		button-sw1 {
+			label = "sw1";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+		};
+
+		button-sw2 {
+			label = "sw2";
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_1>;
+		};
+
+		button-wps {
+			label = "wps";
+			gpios = <&gpio 41 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_lan {
+			label ="green:lan";
+            color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_LAN;
+			gpios = <&gpio 5 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_power: power {
+			label = "green:power";
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_POWER;
+			gpios = <&gpio 46 GPIO_ACTIVE_LOW>;
+		};
+
+		led_usb {
+			label = "green:usb";
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_USB;
+			gpios = <&gpio 43 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&ohci_port1>, <&ehci_port1>;
+			linux,default-trigger = "usbport";
+		};
+
+		led_wan {
+			label = "green:wan";
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_WAN;
+			gpios = <&gpio 39 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wlan {
+			label = "green:wlan";
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_WLAN;
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wps {
+			label = "green:wps";
+			function = LED_FUNCTION_WPS;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "i2c", "i2s", "p0led_an", "p2led_an", "p4led_an", "uart1", "wdt", "wled_an";
+		function = "gpio";
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	mt76@0,0 {
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x28000>;
```

Comment:
Also deprecated.

Reply:
https://github.com/openwrt/openwrt/commit/e93f41adee3e5fa4046811890b92c9b86875911b

Reply:
@kjonca This part can be changed like this:
```
&eeprom_factory_28000 {
	/* tl-wr902ac-v4 has different eeprom size '0x4da8' for MT7613 */
	reg = <0x28000 0x4da8>;
};

&pcie0 {
	wifi@0,0 {
		compatible = "mediatek,mt76";
		reg = <0x0000 0 0 0 0>;
		ieee80211-freq-limit = <5000000 6000000>;
		nvmem-cells = <&eeprom_factory_28000>, <&macaddr_factory_f100>;
		nvmem-cell-names = "eeprom", "mac-address";
		mac-address-increment = <(-1)>;
	};
};
```

Reply:
hm.
```
Error: ../dts/mt7628an_tplink_tl-wr902ac-v4.dts:98.1-22 Label or path eeprom_factory_28000 not found
FATAL ERROR: Syntax error parsing input tree
make[5]: *** [Makefile:235: /home/kjonca/tmp/openwrt/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt76x8/image-mt7628an_tplink_tl-wr902ac-v4.dtb] Error 1
```

Reply:
This change can only be applied to the main branch.

Reply:
Sorry, I do not understand.

Reply:
> Sorry, I do not understand.

Are you compiling this branch?
https://github.com/kjonca/openwrt/tree/tp-link-tplink_tl-wr902ac-v4

I can find the missing label in your error log:

Reply:
> Sorry, I do not understand.

It's possible that the suggested change depends on a feature very recently committed to the code, so your tree needs to be brought up to date (using git rebase) for it to work.

Reply:
Ok, I (probably) understand - I compiled v23.05.0 with my patch. When I upgrade to main - it started to compile.